wallet-new/test/unit/Test/Spec/Addresses.hs

```diff
@@ -408,55 +408,50 @@ spec = describe "Addresses" $ do
-                        let pp2 = PaginationParams (Page 2) (PerPage (quot expectedTotal 3 + 1))
-                        let pp3 = PaginationParams (Page 3) (PerPage (quot expectedTotal 3 + 1))
-                        res <- runExceptT $ runHandler' $ do
-                            Handlers.listAddresses layer (RequestParams pp)
-                        res' <- runExceptT $ runHandler' $ do
-                            Handlers.listAddresses layer (RequestParams pp)
-                        res1 <- runExceptT $ runHandler' $ do
-                            Handlers.listAddresses layer (RequestParams pp1)
-                        res1' <- runExceptT $ runHandler' $ do
-                            Handlers.listAddresses layer (RequestParams pp1)
-                        res2 <- runExceptT $ runHandler' $ do
-                            Handlers.listAddresses layer (RequestParams pp2)
-                        res2' <- runExceptT $ runHandler' $ do
-                            Handlers.listAddresses layer (RequestParams pp2)
-                        res3 <- runExceptT $ runHandler' $ do
-                            Handlers.listAddresses layer (RequestParams pp3)
-                        res3' <- runExceptT $ runHandler' $ do
-                            Handlers.listAddresses layer (RequestParams pp3)
-                        res `shouldBe` res'
-                        res1 `shouldBe` res1'
-                        res2 `shouldBe` res2'
-                        res3 `shouldBe` res3'
+                    withAddressesFixtures 3 8 $ \_ layer _ (_, expectedTotal) -> do
+                        let ppSplit = quot expectedTotal 3 + 1
+                            pp = PaginationParams (Page 1) (PerPage 40)
+                            pp1 = PaginationParams (Page 1) (PerPage ppSplit)
+                            pp2 = PaginationParams (Page 2) (PerPage ppSplit)
+                            pp3 = PaginationParams (Page 3) (PerPage ppSplit)
+                            mkRequest mypp = Handlers.listAddresses layer (RequestParams mypp)
+                        _ <- runExceptT $ runHandler' $ do
+                            r1 <- mkRequest pp
+                            r2 <- mkRequest pp
+                            return $ r1 `shouldBe` r2
+                        _ <- runExceptT $ runHandler' $ do
+                            r1 <- mkRequest pp1
+                            r2 <- mkRequest pp1
+                            return $ r1 `shouldBe` r2
+                        _ <- runExceptT $ runHandler' $ do
+                            r1 <- mkRequest pp2
+                            r2 <- mkRequest pp2
+                            return $ r1 `shouldBe` r2
+                        _ <- runExceptT $ runHandler' $ do
+                            r1 <- mkRequest pp3
+                            r2 <- mkRequest pp3
+                            return $ r1 `shouldBe` r2
```

Comment:
Are you sure this is correct? Here you are returning an `Expectation`, which is just an `Assertion`, which is just an `IO ()` action. But yet, you are now swallowing this result type with `_` which makes me believe you are returning a monadic action _without actually executing it_. Am I right?

Reply:
yes you're right. That's a mistake.